### PR TITLE
🐛 fix: kustomize problem in e2e

### LINKED ISSUE #5165: [e2e] Tests are failing on main
/kind bug

**What steps did you take and what happened:**

The e2e tests are failing on main before it gets to anything to do with the AMIs. Looking at the error its failing when running kustomize here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/shared/suite.go#L79

The error is:

```
Error: trouble configuring builtin PatchTransformer with config: `
patch: platform-kustomization.yaml
`: unable to parse SM or JSON patch from [platform-kustomization.yaml]
```

I created a PR (see #5164) to just run the e2e on main.

**What did you expect to happen:**

I'd expect the e2e to get past this point.

**Anything else you would like to add:**

I don't see what has changed in this area, the upstream function (see this: https://github.com/kubernetes-sigs/cluster-api/blob/main/test/framework/kubernetesversions/template.go#L71) hasn't changed in years.

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by richardcase**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

When running the e2e tests after bumping capi to v1.8.x there is an issue with the e2e tests when we setup the manifests to use for the tests. A package we import from the upstream test framework had been updated to use kustomize v5 format. We where still using v4.

Also, there is a bug in the upstream package and so we have included the package directly with the fix until its merged upstream.

Thanks @damdo  for help with this.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5165 

**Special notes for your reviewer**:

Upstream issue: https://github.com/kubernetes-sigs/cluster-api/pull/11305

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Fix kustomize issue with e2e tests.
```

### CODE REVIEW

**Comment by richardcase** [comment]
the e2e will fail still but it should progress further (and fail on the AMI):

/test pull-cluster-api-provider-aws-e2e

**Comment by richardcase** [comment]
Excellent, the e2e is failing on getting the AMI as expected:

```
 STEP: Searching for AMI: name=capa-ami-ubuntu-18.04-v1.29.1* @ 10/18/24 15:42:41.546
```

I'd like to get this merged if possible. We'll need to follow up on updating the rest of the kustomization.yaml files.

File: test/e2e/shared/suite.go [hunk 1]
```diff
@@ -37,9 +37,9 @@ import (
 	. "github.com/onsi/gomega"
 	"sigs.k8s.io/yaml"
 
+	"sigs.k8s.io/cluster-api-provider-aws/v2/test/helpers/kubernetesversions"
```
Comment (damdo) [inline test/e2e/shared/suite.go:40]: We could add a comment here saying this is a temporary workaround until a new CAPI v1.8.5 with https://github.com/kubernetes-sigs/cluster-api/pull/11307 included

**Review by damdo — COMMENTED** [review]
/lgtm

**Pull request merged**